infra/growth/src/resources/rules.js

```diff
@@ -465,13 +504,33 @@ class BaseRule {
    * @param {string} ethAddress - User's eth address
    * @param {Array<models.GrowthEvent>} events - All events for user since sign up.
    * @param {number} currentUserLevel - Current level the user is at in the campaign.
+   * @param {Arry<object>} rules - All other rules of the campaign
    * @returns {Promise<enums.GrowthActionStatus>}
    */
-  async getStatus(ethAddress, events, currentUserLevel) {
+  async getStatus(ethAddress, events, currentUserLevel, rules) {
     // If the user hasn't reached the level the rule is in, status is Inactive.
     if (currentUserLevel < this.levelId) {
       return GrowthActionStatus.Inactive
     }
+    // If the rule has additional unlock conditions
+    else if (this.config.additionalLockConditions) {
+      this._validateLockConditions(rules, this.config.additionalLockConditions)
```

Comment:
As opposed to calling _validateLockConditions here, could we call it in the constructor of the rule ? This way we wouldn't wait for getStatus to get called to find configuration issues.

Reply:
Yup good point thanks. Moved to the constructor